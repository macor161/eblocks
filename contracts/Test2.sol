pragma solidity ^0.5.0;


contract Test2 {

  uint256 public s;

  constructor() public {
    
  }

  function setS(uint256 _value) public returns(uint256) {
    s = s + _value;
    return s;
  }  

  function setS(string memory _value) public returns(uint256) {
    return 4;
  }   

  /*
  function getToken() public view returns(ERC20) {
    return token;
  }

  function getToken2() public view returns(ERC20 t, uint256 num) {
    t = token;
    num = 6;
  }*/
}

 