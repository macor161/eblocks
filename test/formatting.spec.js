const outdent = require('outdent')
const { formatSource } = require('../src/formatting')


function getMockedColors() {
    return {
        controls: jest.fn(text => text), 
        declarations: jest.fn(text => text), 
        types: jest.fn(text => text), 
        comments: jest.fn(text => text), 
        lineNumbers: jest.fn(text => text)         
    }
}

describe('Fromatting', () => {
    xtest('Format types', async () => {  
        
        const source = outdent`
        contract Test is Test2 {
            uint256 public s3;
          
            constructor(bytes32 _key, address _address) public {
              string memory myString;
            }
          
            function registerContract(bytes32 _key, address _address) public returns (bool) {
              require(_address == address(0));
              int var1 = -32;    
          
              return true;
            }
          }`

        const colors = getMockedColors()
        
        formatSource(source.split('\n'), { colors })

        expect(colors.types.mock.calls.length).toBe(9)
        expect(colors.types.mock.calls[0][0]).toBe('uint256')
        expect(colors.types.mock.calls[1][0]).toBe('bytes32')
        expect(colors.types.mock.calls[2][0]).toBe('address')
        expect(colors.types.mock.calls[3][0]).toBe('string')
        expect(colors.types.mock.calls[4][0]).toBe('bytes32')
        expect(colors.types.mock.calls[5][0]).toBe('address')
        expect(colors.types.mock.calls[6][0]).toBe('bool')
        expect(colors.types.mock.calls[7][0]).toBe('address')
        expect(colors.types.mock.calls[8][0]).toBe('int')
    })

    test('Format declarations', async () => {  
        
        const source = outdent`
        pragma solidity ^0.5.0;
        import "./Test2.sol";

        contract Test is Test2 {          
            constructor(bytes32 _key, address _address) public {
              bool t = false;
            }
          
            function trueFunction() private returns (bool) {
              int var1 = -32;    
          
              return true;
            }
          }`

        const colors = getMockedColors()
        
        formatSource(source.split('\n'), { colors })

        expect(colors.declarations.mock.calls.length).toBe(9)
        
        expect(colors.declarations.mock.calls[0][0]).toBe('solidity')
        expect(colors.declarations.mock.calls[1][0]).toBe('contract')
        expect(colors.declarations.mock.calls[2][0]).toBe('is')
        expect(colors.declarations.mock.calls[3][0]).toBe('constructor')
        expect(colors.declarations.mock.calls[4][0]).toBe('public')
        expect(colors.declarations.mock.calls[5][0]).toBe('false')
        expect(colors.declarations.mock.calls[6][0]).toBe('function')
        expect(colors.declarations.mock.calls[7][0]).toBe('private')
        expect(colors.declarations.mock.calls[8][0]).toBe('true')
    })

    
    xtest('Format single line comments', async () => {  
        
        const source = outdent`
        // This is a test
        function setS(uint256 _value) public returns(uint256) {
          s = s + _value; // This is another test
          return s;
        } `

        const colors = getMockedColors()
        
        formatSource(source.split('\n'), { colors })

        expect(colors.comments.mock.calls.length).toBe(2)
        expect(colors.comments.mock.calls[0][0]).toBe('// This is a test')
        expect(colors.comments.mock.calls[1][0]).toBe('// This is another test')

    })

})

