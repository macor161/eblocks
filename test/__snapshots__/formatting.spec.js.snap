// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error and warning formatting formatError should return the correct formatting 1`] = `
"
[47m[30mERC20.sol[39m[49m
   [1mLine 33:[22m Unused local variable.

  [90m 30 |[39m     * @dev Total number of tokens in existence
  [90m 31 |[39m     */
  [90m 32 |[39m     [34mfunction[39m totalSupply() [34mpublic[39m view [35mreturns[39m ([36muint256[39m) {
[1m[31m>[39m[22m [90m 33 |[39m         [36muint256[39m t = 3;
  [90m    |[39m [1m[31m        ^[39m[22m
  [90m 34 |[39m         [35mreturn[39m _totalSupply;
  [90m 35 |[39m     }
"
`;

exports[`Error and warning formatting formatErrors should return the correct formatting 1`] = `
"
[47m[30mERC20.sol[39m[49m
   [1mLine 33:[22m Unused local variable.

  [90m 30 |[39m     * @dev Total number of tokens in existence
  [90m 31 |[39m     */
  [90m 32 |[39m     [34mfunction[39m totalSupply() [34mpublic[39m view [35mreturns[39m ([36muint256[39m) {
[1m[31m>[39m[22m [90m 33 |[39m         [36muint256[39m t = 3;
  [90m    |[39m [1m[31m        ^[39m[22m
  [90m 34 |[39m         [35mreturn[39m _totalSupply;
  [90m 35 |[39m     }

[47m[30mTest2.sol[39m[49m
   [1mLine 15:[22m Unused function parameter. Remove or comment out the variable name to silence this warning.

  [90m 12 |[39m     [35mreturn[39m s;
  [90m 13 |[39m   }  
  [90m 14 |[39m 
[1m[31m>[39m[22m [90m 15 |[39m   [34mfunction[39m setS([36mstring[39m [34mmemory[39m _value) [34mpublic[39m [35mreturns[39m([36muint256[39m) {
  [90m    |[39m [1m[31m                ^[39m[22m
  [90m 16 |[39m     [35mreturn[39m 4;
  [90m 17 |[39m   }   

[47m[30mTest2.sol[39m[49m
   [1mLine 15:[22m Function state mutability can be restricted to pure

  [90m 12 |[39m     [35mreturn[39m s;
  [90m 13 |[39m   }  
  [90m 14 |[39m 
[1m[31m>[39m[22m [90m 15 |[39m   [34mfunction[39m setS([36mstring[39m [34mmemory[39m _value) [34mpublic[39m [35mreturns[39m([36muint256[39m) {
  [90m    |[39m [1m[31m  ^[39m[22m
  [90m 16 |[39m     [35mreturn[39m 4;
  [90m 17 |[39m   }   

[1m[31mFailed to compile: 3 error(s) found.[39m[22m
"
`;

exports[`Error and warning formatting formatWarning should return the correct formatting 1`] = `
"
[47m[30mERC20.sol[39m[49m
   [1mLine 33:[22m Unused local variable.

  [90m 30 |[39m     * @dev Total number of tokens in existence
  [90m 31 |[39m     */
  [90m 32 |[39m     [34mfunction[39m totalSupply() [34mpublic[39m view [35mreturns[39m ([36muint256[39m) {
[1m[33m>[39m[22m [90m 33 |[39m         [36muint256[39m t = 3;
  [90m    |[39m [1m[33m        ^[39m[22m
  [90m 34 |[39m         [35mreturn[39m _totalSupply;
  [90m 35 |[39m     }
"
`;

exports[`Error and warning formatting formatWarnings should return the correct formatting 1`] = `
"
[47m[30mERC20.sol[39m[49m
   [1mLine 33:[22m Unused local variable.

  [90m 30 |[39m     * @dev Total number of tokens in existence
  [90m 31 |[39m     */
  [90m 32 |[39m     [34mfunction[39m totalSupply() [34mpublic[39m view [35mreturns[39m ([36muint256[39m) {
[1m[33m>[39m[22m [90m 33 |[39m         [36muint256[39m t = 3;
  [90m    |[39m [1m[33m        ^[39m[22m
  [90m 34 |[39m         [35mreturn[39m _totalSupply;
  [90m 35 |[39m     }

[47m[30mTest2.sol[39m[49m
   [1mLine 15:[22m Unused function parameter. Remove or comment out the variable name to silence this warning.

  [90m 12 |[39m     [35mreturn[39m s;
  [90m 13 |[39m   }  
  [90m 14 |[39m 
[1m[33m>[39m[22m [90m 15 |[39m   [34mfunction[39m setS([36mstring[39m [34mmemory[39m _value) [34mpublic[39m [35mreturns[39m([36muint256[39m) {
  [90m    |[39m [1m[33m                ^[39m[22m
  [90m 16 |[39m     [35mreturn[39m 4;
  [90m 17 |[39m   }   

[47m[30mTest2.sol[39m[49m
   [1mLine 15:[22m Function state mutability can be restricted to pure

  [90m 12 |[39m     [35mreturn[39m s;
  [90m 13 |[39m   }  
  [90m 14 |[39m 
[1m[33m>[39m[22m [90m 15 |[39m   [34mfunction[39m setS([36mstring[39m [34mmemory[39m _value) [34mpublic[39m [35mreturns[39m([36muint256[39m) {
  [90m    |[39m [1m[33m  ^[39m[22m
  [90m 16 |[39m     [35mreturn[39m 4;
  [90m 17 |[39m   }   
   
[1m[33mCompiled with 3 warning(s)[39m[22m
"
`;
